test/thunkify.js

```diff
@@ -0,0 +1,28 @@
+var R = require('..');
+var eq = require('./shared/eq');
+
+describe('thunkify', function() {
+  it('returns a function with the same arity as the given function', function() {
+    var input = function input(a0, a1) { };
+    var f = R.thunkify(input);
+    eq(typeof f, 'function');
+    eq(f.length, input.length);
+  });
+
+  it('returns a function that expects arguments and returns a new invoker function', function() {
+    var input = function input(a0, a1) { };
+    var f = R.thunkify(input);
+    eq(typeof f(42, 'xyz'), 'function');
+  });
+
+  it('calls the original function with the provided arguments', function() {
+    var f = R.thunkify(R.add(2));
+    eq(f(42)(), 44);
+  });
+
+  it('should support sending partial arguments', function() {
+    var input = function input(a0, a1) { return [a0, a1]; };
+    var f = R.thunkify(input);
+    eq(f(42)(), [42, undefined]);
+  });
```

Comment:
We should add a property based test for this as well. 

`forall f, a, b :: f(a, b) == thunkify(f)(a, b)()`

Reply:
I'm all for adding additional tests, but unsure what you mean by "property based"?

Reply:
Property tests are for verifying invariant properties of a function. There are certain things that should be true about a function regardless of the inputs. With unit tests, we are only testing certain examples, but property tests can throw a bunch of random input at the function and make sure those invariants hold.

[Here](https://hackernoon.com/property-based-testing-4330e3e77381) is a really simple overview with JS examples.

[Here](https://github.com/ramda/ramda/blob/master/test/compose.js#L63) is an example in our codebase of some property tests.

Reply:
I do like the idea of adding property-based testing, but I wouldn't hold up a PR for that.  I would rather do a focused pass through the code to bring it about.